syft/pkg/cataloger/catalog.go

```diff
@@ -69,6 +69,9 @@ func Catalog(resolver source.FileResolver, release *linux.Release, catalogers ..
 
 			// generate PURL (note: this is excluded from package ID, so is safe to mutate)
 			p.PURL = pkg.URL(p, release)
+			if p.Language == "" {
```

Comment:
nit: an extra comment explaining why this is here would be good since there is some high-level reasoning for it that isn't apparent.